src/sonic-ctrmgrd/ctrmgr/readiness_probe.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# This script is used by k8s to check the readiness of containers
+# Check if the container is readiness or not, exit code 0 means readiness, others mean not readiness
+
+#### exit code contract, k8s only cares zero or not none-zero, but we want to use none-zero code to indicate different error
+# 0: readiness
+# 1: python script crach exit code
+# 2: supervisor start service doesn't exit normally
+# other exit code: returned by post_check_script, define in the post_check_script, should not include 1,2
+
+# check if the start service exists
+# if the start service exists, check if it exits normally
+# if the start service doesn't exist normally, exit with code 2
+pre_check_service_name="start"
+supervisorctl status |awk '{print $1}' |grep -w $pre_check_service_name > /dev/null
+start_check_result=$?
+if [ $start_check_result = 0 ] && [ $(supervisorctl status $pre_check_service_name |awk '{print $2}') != 'EXITED' ]; then
+    exit 2
+fi
+
+# feature owner can add their own readiness check script
+# check if the post_check_script exists
+# if the post_check_script exists, run it
+# if the post_check_script exits with non-zero code, exit with the code
+post_check_script="/usr/bin/readiness_probe.py"
```

Comment:
>/usr/bin/readiness_probe.py [](http://example.com/codeflow?start=19&length=27)

Do not assume python3.
How about `/usr/bin/readiness_probe_hook`. 
```
if [ -x $post_check_script ]; then
    $post_check_script
```

Reply:
I think although we are defining the contract, the hook script here should be more specific. By the way, are there some scenarios that we can't use python code as the hook script here. I think a specific language here will more definite, such as the script crash code, python crash code is 1. We don't need to know other languages crash code.